scripts/Diffraction/isis_powder/routines/sample_details.py

```diff
@@ -4,21 +4,28 @@
 from isis_powder.routines import common
 import math
 
+property_err_string = "The following sample property was not passed as an argument: "
+
 
 class SampleDetails(object):
     def __init__(self, **kwargs):
-        # By using kwargs we get a better error that, init takes 4 arguments
-        err_string = "A following sample property was not passed as an argument: "
-        height = common.dictionary_key_helper(dictionary=kwargs, key="height", exception_msg=err_string + "height")
-        radius = common.dictionary_key_helper(dictionary=kwargs, key="radius", exception_msg=err_string + "radius")
-        center = common.dictionary_key_helper(dictionary=kwargs, key="center", exception_msg=err_string + "center")
-
-        # Currently we only support cylinders
-        self.shape_type = "cylinder"
-        SampleDetails._validate_sample_details_constructor_inputs(height=height, radius=radius, center=center)
-        self.height = float(height)
-        self.radius = float(radius)
-        self.center = [float(i) for i in center]  # List of X, Y, Z position
+        self.shape_type = common.dictionary_key_helper(dictionary=kwargs, key="shape",
+                                                       exception_msg="Sample property \"shape\" was not passed to "
+                                                                     "SampleDetails. If you have been using this class "
+                                                                     "without worrying about \"shape\", you probably "
+                                                                     "want to set this to \"cylinder\"")
+        center = common.dictionary_key_helper(dictionary=kwargs, key="center",
+                                              exception_msg=property_err_string + "center")
+        SampleDetails._validate_center(center)
+        self._center = [float(i) for i in center]  # List of X, Y, Z position
+
+        if self.shape_type == "cylinder":
```

Comment:
`shape_type` and `shape` should be called `_shape_type` and `_shape` to match the [python conventions](https://www.python.org/dev/peps/pep-0008/#descriptive-naming-styles).